driver/tests/mpi/test_restart.py

```diff
@@ -0,0 +1,111 @@
+import subprocess
+import unittest.mock
+
+import gt4py
+import numpy as np
+import pytest
+import xarray as xr
+import yaml
+import zarr
+
+import pace.dsl
+from pace.driver import CreatesComm, DriverConfig
+from pace.driver.state import DriverState
+from pace.util.null_comm import NullComm
+
+
+class NullCommConfig(CreatesComm):
+    def __init__(self, layout):
+        self.layout = layout
+
+    def get_comm(self):
+        return NullComm(
+            rank=0,
+            total_ranks=6 * self.layout[0] * self.layout[1],
+            fill_value=0.0,
+        )
+
+    def cleanup(self, comm):
+        pass
+
+
+def test_default_save_restart():
+    driver_config = DriverConfig(
+        stencil_config=pace.dsl.StencilConfig(),
+        nx_tile=12,
+        nz=79,
+        dt_atmos=225,
+        days=0,
+        hours=0,
+        minutes=0,
+        seconds=225,
+        layout=(1, 1),
+        initialization=unittest.mock.MagicMock(),
+        performance_config=unittest.mock.MagicMock(),
+        comm_config=NullCommConfig((1, 1)),
+        diagnostics_config=unittest.mock.MagicMock(),
+        dycore_config=unittest.mock.MagicMock(nwat=6),
+        physics_config=unittest.mock.MagicMock(),
+    )
+    assert driver_config.save_restart is False
+
+
+def test_restart_results():
+    subprocess.call("tests/mpi/run_save_and_load_restart.sh")
+    restart = xr.open_zarr(
+        store=zarr.DirectoryStore(path="output.zarr"), consolidated=False
+    )
+    regular = xr.open_zarr(
+        store=zarr.DirectoryStore(path="run_two_steps_output.zarr"), consolidated=False
+    )
+    assert restart["time"][0] == regular["time"][-1]
+    for var in [
+        "u",
+        "v",
+        "ua",
+        "va",
+        "pt",
+        "delp",
+        "qvapor",
+        "qliquid",
+        "qice",
+        "qrain",
+        "qsnow",
+        "qgraupel",
+    ]:
+        np.testing.assert_allclose(
+            restart[var].isel(time=0).values, regular[var].isel(time=-1).values
+        )
+
+
+@pytest.mark.last
+def test_driver_state_load_restart():
```

Comment:
It's pretty finnicky to rely on artifacts from a previous test to be used in a later test. Can you combine this with the test that runs the model? It can be used as a helper function if that's easier.

Reply:
Merged to one